### PR TITLE
fix(manager): normalize cwd before gate keying and cache lookups

### LINKED ISSUE #24: Normalize cwd before gate keying and conversation cache lookups
- [ ] `keyFor` builds the gate key from the raw client-supplied cwd (internal/manager/concurrency.go:83-91) and resolveLatest/snapshotCwd/captureNewUUID do raw map lookups (internal/manager/conversation.go:21-39). `/x/y/`, a relative path, or a symlinked alias of one directory gets a distinct gate key, so two "same dir" fresh runs are not serialized (the agy session-lock scenario the gate exists to prevent) and the agy cache entry is missed (`continue_latest` silently starts a new conversation, fresh-run capture never matches).
- [ ] `readSessions` already applies filepath.Clean to both sides (internal/manager/sessions.go:46-52), so normalization exists in one place and is missing in the other three.
- [ ] StartJob swallows os.Getwd failure (manager.go:137-142): cwd "" produces gate key "", which skips serialization entirely and runs agy in an inherited directory. Should be an error.
- [ ] While in there: StartJob writes the resolved cwd back into req but keeps resolved model/timeout as separate locals while req.Model/req.Timeout stay stale (manager.go:143-152, buildAgyArgs at 404-417). Normalize all or none, so a future use of req.Model cannot silently bypass the default fallback.

Fix: filepath.Clean (and best-effort EvalSymlinks) once at the top of StartJob; use the result for the key, the snapshot, cmd args, and persisted meta. Regression test with a trailing-slash cwd.

### PR DESCRIPTION
**Pull request by tphakala**
## Summary

A raw client-supplied `cwd` flowed straight into the gate key and the agy conversation-cache lookups, so a trailing slash, a relative path, or a symlinked alias of one directory produced a distinct gate key. Two "same dir" fresh runs were then not serialized (re-exposing the agy session-lock hang the gate exists to prevent), and the cache entry was missed (`continue_latest` silently started a new conversation; fresh-run id capture never matched).

The fix canonicalizes the cwd once at the top of `StartJob` with `filepath.Abs` plus a best-effort `filepath.EvalSymlinks`, and uses the result for the gate key, the cache snapshot/resolve, the agy args, and the persisted meta. Because the supervisor sets `cmd.Dir` to this value, agy's own `getcwd` returns the symlink-resolved physical path and keys `last_conversations.json` by that, so resolving symlinks makes our key match agy's.

### Changes
- **Normalize cwd in `StartJob`** and feed it to the gate key, cache lookups, args, and meta.
- **Fail `StartJob` when `os.Getwd` errors** instead of running agy in an inherited directory under gate key `""`.
- **Resolve `model`/`timeout` defaults back into `req`** so the gate key, args, and meta all derive from one normalized request; a later read of a stale `req.Model`/`req.Timeout` can no longer bypass the default fallback. `buildAgyArgs` now reads from `req`.
- **Canonicalize the `list_sessions` filter** through the same helper, so a symlinked or relative filter matches the resolved paths agy keys its cache by (previously it never matched).
- **Re-normalize cwd in `reqFromMeta`** so a job persisted by an older binary (raw cwd) restores under the same key a new normalized run computes, closing an upgrade-window race where the two would not serialize and could issue concurrent `O_TRUNC` writes to the same agy cache entry.

The last two items address findings from peer review (five parallel reviewers plus Gemini 3.1 Pro cross-validation) that confirmed the core fix and surfaced the `list_sessions` inconsistency and the upgrade-window race.

## Related Issues
Closes #24

## Test Plan
- [x] `go test ./...` passes
- [x] `golangci-lint run ./...` clean, `gofmt`/`go vet` clean
- [x] `go build -tags ruleguard ./rules/` builds
- [x] New regression tests cover: trailing-slash / relative / symlinked spellings, the unresolvable-path fallback, the normalized persisted meta, two equivalent-cwd runs collapsing to one gate key (verified to fail without the fix), the symlinked `list_sessions` filter (verified to fail without the fix), and the legacy-meta restore key (verified to fail without the fix)

## Summary by CodeRabbit

* **Bug Fixes**
  * Working directory paths are consistently normalized for job starts, restores, gating, and session reads so equivalent CWD spellings (symlinks, trailing slashes, relative forms) behave identically.
  * Job startup records consistent defaults (model/timeout) in job metadata and reliably rejects conflicting concurrent runs for the same canonical CWD.

* **Tests**
  * Added regression tests covering CWD normalization, job serialization conflicts, session filtering with symlinks, and related cleanup to keep tests deterministic.

### CODE REVIEW

File: internal/manager/job_linux_test.go [hunk 1]
```diff
@@ -91,6 +92,99 @@ func TestStartJobCleansUpDirOnSpawnFailure(t *testing.T) {
...
+	if err != nil {
+		t.Fatalf("first StartJob: %v", err)
+	}
+	// Reap the whole supervisor group (bash + the sleeping fake agy) promptly,
+	// regardless of how the test exits; the fake supervisor does not forward
+	// signals, so Cancel alone would leave the sleep running.
+	t.Cleanup(func() {
+		if meta, lerr := m.store.Load(job1.ID); lerr == nil {
+			_ = syscall.Kill(-meta.PID, syscall.SIGKILL)
+		}
```
[the remark this answers is not in the document]
Comment (tphakala) [reply internal/manager/job_linux_test.go]: Good catch, fixed in eaf9721. Guarded `meta.PID > 0` before `syscall.Kill(-meta.PID, ...)` so a zero-value PID from a failed Load can't signal the test runner's own process group.

File: internal/manager/manager.go [hunk 2]
```diff
@@ -247,20 +247,32 @@ func (m *Manager) StartJob(req StartRequest) (Job, error) {
...
-		model = m.cfg.DefaultModel
-	}
-	timeout := req.Timeout
-	if timeout <= 0 {
-		timeout = m.cfg.DefaultTimeout
+	// Canonicalize the cwd once so the gate key, cache lookups, cmd.Dir, and
+	// persisted meta all share one spelling (see normalizeCwd).
+	cwd, err = normalizeCwd(cwd)
+	if err != nil {
+		return Job{}, fmt.Errorf("normalize working directory %q: %w", req.Cwd, err)
```
[the remark this answers is not in the document]
Comment (tphakala) [reply internal/manager/manager.go]: Fixed in eaf9721, with a tweak: `normalizeCwd` returns `("", err)` on failure, so reading the post-call `cwd` would also print empty. I capture the pre-normalization input instead (`normalized, err := normalizeCwd(cwd)` then report `cwd`), so the actual directory that failed is shown regardless of whether it came from the request or `os.Getwd`.

File: internal/manager/manager.go [hunk 3]
```diff
@@ -457,8 +469,20 @@ func (m *Manager) runPeriodicGC(ctx context.Context, interval time.Duration) {
...
+// fail to serialize, and risk concurrent O_TRUNC writes to the same agy cache
+// entry. For jobs started by the current binary meta.Cwd is already normalized,
+// so this is idempotent.
 func reqFromMeta(meta jobstore.Meta) StartRequest {
-	return StartRequest{ConversationID: meta.ConversationID, Cwd: meta.Cwd}
+	cwd := meta.Cwd
+	if norm, err := normalizeCwd(cwd); err == nil {
+		cwd = norm
+	}
+	return StartRequest{ConversationID: meta.ConversationID, Cwd: cwd}
```
[the remark this answers is not in the document]
Comment (tphakala) [reply internal/manager/manager.go:494]: Fixed in eaf9721 at the root: `normalizeCwd` now returns `""` for an empty input rather than letting `filepath.Abs("")` resolve to the process cwd, so `reqFromMeta` preserves the original no-key behavior for a legacy empty `meta.Cwd`. Added a regression test (`TestReqFromMetaPreservesEmptyCwd`).

File: internal/manager/normalize.go [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+	if err != nil {
+		return "", err
+	}
+	// Best-effort symlink resolution: a path that does not exist yet (agy will
+	// fail on it regardless) or is otherwise unresolvable keeps the cleaned
+	// absolute form instead of failing the run here.
+	if resolved, err := filepath.EvalSymlinks(abs); err == nil {
+		return resolved, nil
+	}
+	return abs, nil
```
[the remark this answers is not in the document]
Comment (tphakala) [reply internal/manager/normalize.go:34]: Fixed in eaf9721 exactly as suggested: added the empty-cwd guard at the top of `normalizeCwd`. Also added regression tests for both `normalizeCwd("")` and `reqFromMeta` with an empty cwd, and confirmed they fail without the guard (the old path resolved to the package directory).

**Comment by tphakala** [comment]
@coderabbitai review

File: internal/manager/manager.go [hunk 5]
```diff
@@ -457,8 +471,20 @@ func (m *Manager) runPeriodicGC(ctx context.Context, interval time.Duration) {
...
+// legacy job (raw cwd) and a new run (normalized cwd) would hold different keys,
+// fail to serialize, and risk concurrent O_TRUNC writes to the same agy cache
+// entry. For jobs started by the current binary meta.Cwd is already normalized,
+// so this is idempotent.
 func reqFromMeta(meta jobstore.Meta) StartRequest {
-	return StartRequest{ConversationID: meta.ConversationID, Cwd: meta.Cwd}
+	cwd := meta.Cwd
+	if norm, err := normalizeCwd(cwd); err == nil {
+		cwd = norm
+	}
```
[the remark this answers is not in the document]
Comment (tphakala) [reply internal/manager/manager.go:493]: Good point on the inconsistency, fixed in 7475b3f. I kept reqFromMeta best-effort (RestoreGate has to stay resilient and can't fail closed the way StartJob does), but added a log on the fallback so an operator is alerted that a restored job's gate key may not match a new same-directory run under that rare path (legacy relative cwd plus an os.Getwd failure).
